Add brew installation option to documentation. (#818)
fermyon/developer: content/spin/install.md

{{ startTab "Linux"}}

> You can now manage your Spin installation via [Homebrew](https://brew.sh/). Homebrew automatically installs Spin templates and Spin plugins, and on uninstall, will prompt you to delete the directory where the templates and plugins were downloaded:

Reviewer: I'd suggest removing the 'now' as this speaks to this very moment :-)

Author: Thanks, done.

> You can now manage your Spin installation via [Homebrew](https://brew.sh/). Homebrew automatically installs Spin templates and Spin plugins, and on uninstall, will prompt you to delete the directory where the templates and plugins were downloaded:

Reviewer 2: We should say what that directory is (where plugins and templates are installed), namely `$HOMEBREW_PREFIX/fermyon-spin`

Reviewer 3: Agree though this would be best done i https://developer.fermyon.com/spin/cache rather than here

> You can now manage your Spin installation via [Homebrew](https://brew.sh/). Homebrew automatically installs Spin templates and Spin plugins, and on uninstall, will prompt you to delete the directory where the templates and plugins were downloaded:

Reviewer 2: There is a [homebrew core formula for a verification tool called spin](https://formulae.brew.sh/formula/spin). We should call out that a `brew install spin` will not install our Spin. They need to install the spin from our tap. (another reason i recommend breaking out the steps into a tap install and then a project install to emphasize the tap).

Author: Thanks, note added.

> <pre class="bash spin-install" id="spin-install-linux"><code>$ brew tap fermyon/tap && brew install fermyon/tap/spin</code></pre>

Reviewer 2: I'd recommend breaking this into two steps to emphasize that they first need to install the Fermyon tap, a repository of all the Fermyon Homebrew packages, and then isntall spin with `fermyon/tap/spin`. See Hashicorp for example: https://developer.hashicorp.com/nomad/docs/install.

We also should add an instruction on how to upgrade to latest:
`brew upgrade fermyon/tap/nomad`

Reviewer 3: Again, I agree about documenting the upgrade command, but with the current docs structure this would go better on https://developer.fermyon.com/spin/upgrade

Author: Please see new PR that updates the upgrade page
https://github.com/fermyon/developer/pull/827

> <pre class="bash spin-install" id="spin-install-linux"><code>$ brew tap fermyon/tap && brew install fermyon/tap/spin</code></pre>

Author: ```suggestion
Add Spin to the list of formulae that Homebrew tracks, updates, and installs from:

<pre class="bash spin-install" id="spin-install-linux"><code>$ brew tap fermyon/tap</code></pre>

Install Spin:

<pre class="bash spin-install" id="spin-install-linux"><code>$ brew install fermyon/tap/spin</code></pre>

> Note: `brew install spin` will **not** install Fermyon's Spin framework. Fermyon Spin is accessed from the `fermyon` tap, as shown above.

{{ startTab "Linux"}}

You can manage your Spin installation via [Homebrew](https://brew.sh/). Homebrew automatically installs Spin templates and Spin plugins, and on uninstall, will prompt you to delete the directory where the templates and plugins were downloaded:

> Add Spin to the list of formulae that Homebrew tracks, updates, and installs from:

Reviewer 2: technically that is not what this command does. Rather it fetches Fermyon's homebrew tap
```suggestion
Install the Fermyon tap, which Homebrew tracks, updates, and installs Spin from:
```

{{ startTab "macOS"}}

> You can now manage your Spin installation via [Homebrew](https://brew.sh/). Homebrew automatically installs Spin templates and Spin plugins, and on uninstall, will prompt you to delete the directory where the templates and plugins were downloaded:

Reviewer: I'd suggest removing the 'now' as this speaks to this very moment :-)

Author: Thanks, done.

> <pre class="bash spin-install" id="spin-install-macos"><code>$ brew tap fermyon/tap && brew install fermyon/tap/spin</code></pre>

Author: ```suggestion
Add Spin to the list of formulae that Homebrew tracks, updates, and installs from:

<pre class="bash spin-install" id="spin-install-macos"><code>$ brew tap fermyon/tap</code></pre>

Install Spin:

<pre class="bash spin-install" id="spin-install-macos"><code>$ brew install fermyon/tap/spin</code></pre>

> Note: `brew install spin` will **not** install Fermyon's Spin framework. Fermyon Spin is accessed from the `fermyon` tap, as shown above.

{{ startTab "macOS"}}

You can manage your Spin installation via [Homebrew](https://brew.sh/). Homebrew automatically installs Spin templates and Spin plugins, and on uninstall, will prompt you to delete the directory where the templates and plugins were downloaded:

> Add Spin to the list of formulae that Homebrew tracks, updates, and installs from:

Reviewer 2: ```suggestion
Install the Fermyon tap, which Homebrew tracks, updates, and installs Spin from:
```
